### PR TITLE
Dancer buff + Bug Fixes

### PR DESCRIPTION
**Pull request by Git-Nivrak**
# About the pull request
Removes the mini cooldown on missing dancer impale and tail trip

# Explain why it's good for the game
Will make more people play dancer, Makes it feel smoother to play.
Having your entire combo be interrupted by missing one click is extremely awkward, especially as a t3 with little health and armor being punished for that seems unreasonable.

If this turns out to make dancer too strong the cooldown can be reduced instead of completely removed.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
balance: Removed the cooldown on missing Dancer's impale and tail trip abilities.
fix: Dancer's slash tag no longer gets consumed by tail trip if the target is out of range.
/:cl:

### CODE REVIEW

**Review by morrowwolf — COMMENTED** [review]
I'm happy to try this. I'm not sold on it yet.

File: code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm [hunk 1]
```diff
@@ -714,6 +700,15 @@
 	dancer_user.face_atom(target_carbon)
 	dancer_user.flick_attack_overlay(target_carbon, "disarm")
 
+	if (dancer_user.mutation_type == PRAETORIAN_DANCER)
+		var/found = FALSE
+		for (var/datum/effects/dancer_tag/dancer_tag_effect in target_carbon.effects_list)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm:705]: Use the locate(type) in list syntax instead of a loop

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by TeDGamer** [comment]
For changelog, it's "fix:" not "bugfix:"

**Review by morrowwolf — APPROVED** [review]
Usual disclaimer: if this causes issues it'll be changed blah blah blah.

TM a bit ago when decent without an crazy change in usage or becoming problematic.

Let's get it merged.

**Comment by Ammoniacres** [comment]
This is merged but doesn't seem to work anymore? Removing the cooldown override codelines actually produces the intended effect of removing the cooldowns on miss-click. Should probably be logged as a bug.

**Comment by Git-Nivrak** [comment]
Nice necropost
The code worked it was just reverted in a later pr

**Pull request merged**